code/controller/controldelegator.cpp

```diff
@@ -0,0 +1,41 @@
+#include "controldelegator.h"
```

Comment:
Both actuator and simulator extend controller, and previously there was a controller pointer which could point to anything extending controller, actuator, simulator, or otherwise. I've reverted things back to how they were before